### PR TITLE
Disable image tag support for template collection reference

### PR DESCRIPTION
**Pull request by moria97**
## Description
Support digest only in template collection reference as image tags are mutable and can lead to unexpected results:

- We have added the pulled image to local cache for 30 minutes and when user modified the same tag, the image will not be refreshed until cache expires
- When another admin/user changes the image tag, the conversion result would be strange. 

The decision is to remove support for image tags for now and request with payload of image tags will result in 400 Bad Request ```Digest must be provided in image reference```

## Related issues
Addresses [AB#78458](https://microsofthealth.visualstudio.com/Health/_workitems/edit/78485).

## Testing
Add cases for image tags and digest.

## FHIR Team Checklist
- [ ] **Update the title** of the PR to be succinct and less than 50 characters
- [ ] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [ ] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [ ] Tag the PR with **Azure API for FHIR** if this will release to the managed service
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Skip (fix bug)

### CODE REVIEW

File: src/Microsoft.Health.Fhir.Shared.Web/appsettings.json [hunk 1]
```diff
@@ -71,7 +71,6 @@
                 "Enabled": false,
                 "ContainerRegistryServers": [],
                 "TemplateCollectionOptions": {
-                    "ShortCacheTimeSpan": "00:30:00",
```
Comment (LTA-Thinking) [inline src/Microsoft.Health.Fhir.Shared.Web/appsettings.json:74]: Why are we removing this setting?

**Comment by LTA-Thinking** [comment]
I'm concerned about this being a breaking change. I don't know how many users have custom image repos set up, but this would be a breaking change for them.

**Comment by LTA-Thinking** [comment]
As a bug fix I think not having a semver command is more appropriate than a skip command.

File: src/Microsoft.Health.Fhir.Core/Messages/ConvertData/ConvertDataRequest.cs [hunk 2]
```diff
@@ -58,8 +58,8 @@ public class ConvertDataRequest : IRequest<ConvertDataResponse>
 
         /// <summary>
         /// Reference for template collection.
-        /// The format is "<registryServer>/<imageName>:<imageTag>" for template collection stored in container registries.
-        /// Also supports image digest as reference. Will use 'latest' if no tag or digest present.
+        /// It can be the default templates "microsofthealth/fhirconverter:default", or an image on Azure Container Registry with digest "<registryServer>/<imageName>@<imageDigest>".
```
Comment (brendankowitz) [inline src/Microsoft.Health.Fhir.Core/Messages/ConvertData/ConvertDataRequest.cs:61]: Is `:default` a tag? Is this the only tag that is supported?

Comment (brendankowitz) [reply src/Microsoft.Health.Fhir.Core/Messages/ConvertData/ConvertDataRequest.cs:61]: Oh I think I get it. It must be exactly `microsofthealth/fhirconverter:default` or a reference with an image digest.

File: test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ConvertDataTests.cs [hunk 3]
```diff
@@ -135,9 +135,31 @@ public async Task GivenAValidRequest_ButTemplateReferenceIsInvalid_WhenConvertDa
...
         [InlineData("fakeacr.azurecr.io/template:default")]
         [InlineData("test.azurecr-test.io/template:default")]
-        [InlineData("test.azurecr.com/template@sha256:592535ef52d742f81e35f4d87b43d9b535ed56cf58c90a14fc5fd7ea0fbb8696")]
         [InlineData("*****####.com/template:default")]
         [InlineData("¶Š™œãý£¾.com/template:default")]
+        public async Task GivenAValidRequest_WithoutDigest_WhenConvertData_ShouldReturnBadRequest(string templateReference)
+        {
+            if (!_convertDataEnabled)
+            {
+                return;
```
Comment (brendankowitz) [inline test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ConvertDataTests.cs:144]: We can now use [SkippableTheory/Facts](https://github.com/AArnott/Xunit.SkippableFact/blob/master/test/Xunit.SkippableFact.Tests/SampleTests.cs#L51)

Comment (moria97) [reply test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/ConvertDataTests.cs:144]: Awesome! Thanks!

**Comment by moria97** [comment]
Close this pull request as we may want to keep tags.

**Pull request closed without merging**